src/App.tsx

```diff
@@ -20,17 +25,147 @@ class App extends React.Component<{}, State> {
+  };
+
+  setFrameSize = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ frameSize: +e.target.value });
+  };
+
+  setItemWidth = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ itemWidth: +e.target.value });
+  };
+
+  setAnimationDuration = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ animationDuration: +e.target.value });
+  };
+
+  setInfinite = (e: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ infinite: e.target.checked });
   };
 
   render() {
-    const { images } = this.state;
+    const {
+      images,
+      step,
+      frameSize,
+      itemWidth,
+      animationDuration,
+      infinite,
+    } = this.state;
 
     return (
       <div className="App">
         {/* eslint-disable-next-line */}
         <h1>Carousel with {images.length} images</h1>
+        <h1
+          className="App__title"
+          data-cy="title"
+        >
+          {`Carousel with ${images.length} images`}
+        </h1>
+
+        <form action="/">
+          <label
+            className="form__label"
+            htmlFor="itemId"
+          >
+            Item width
+            <input
+              type="number"
+              className="form__input"
+              name="itemWidth"
+              id="itemId"
+              min={130}
+              max={260}
+              value={itemWidth}
+              onChange={this.setItemWidth}
+            />
+          </label>
+
+          <label
+            className="form__label"
+            htmlFor="frameId"
+          >
+            Frame Size
+            <input
+              type="number"
+              className="form__input"
+              name="frameSize"
+              id="fromeId"
+              min={1}
+              max={images.length}
+              step={1}
+              value={frameSize}
+              onChange={this.setFrameSize}
+            />
+          </label>
+
+          <label
+            className="form__label"
+            htmlFor="stepId"
+          >
+            Step
+            <input
+              type="number"
+              className="form__input"
+              name="step"
+              id="stepId"
+              min={1}
+              max={images.length}
+              step={1}
+              value={step}
+              onChange={this.setStep}
+            />
+          </label>
+
+          <label
+            className="form__label"
+          >
+            Animation Duration
+            <input
+              type="number"
+              className="form__input"
+              name="animationDuration"
+              id="animationDurationId"
+              min={500}
+              max={3000}
+              step={500}
+              value={animationDuration}
+              onChange={this.setAnimationDuration}
+            />
+          </label>
+
+          <label
+            className="form__label"
+          >
+            Infinite
+            <input
+              type="number"
```

Comment:
```suggestion
              type="checkbox"
```
infinite should have type checkbox